list-objects-filter-profile.h

```diff
@@ -0,0 +1,100 @@
+#ifndef GIT_LIST_OBJECTS_FILTER_PROFILE_H
+#define GIT_LIST_OBJECTS_FILTER_PROFILE_H
+
+/**
+ * The List Objects Filter Profile API can be used to develop filter plugins used for
+ * git-upload-pack/git-rev-list/etc.
+ *
+ * See contrib/profile-filters/README.md for more details and examples.
```

Comment:
Yeah - I haven't got to that bit yet. I just wanted to share what I have so far since -
- you said you were thinking about this stuff too, and I didn't want to duplicate effort
- to make sure I'm in the right general direction